### PR TITLE
Change typedefs from @api to @doc

### PR DESCRIPTION
**Pull request by probins**
As discussed in #5416.

Simply copying the current api plugin to another one called doc seems to work just fine. In fact, it's so simple, I'm sure there must be something wrong :-) I regenerated the docs, and all seems to be exactly as before.

If people agree with this, I'll change the olx typedefs as well. And then look into enums. Anything else that should not be `@api`?

### CODE REVIEW

**Comment by gberaudo** [comment]
Is copying the recommended way to achieve this?

File: config/jsdoc/api/plugins/doc.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
+/**
+ * Define an @doc tag
+ */
+var conf = env.conf.stability;
+var defaultLevels = ["deprecated","experimental","unstable","stable","frozen","locked"];
```
Comment (ahocevar) [inline config/jsdoc/api/plugins/doc.js]: I think we don't need to handle levels here.

Comment (probins) [inline config/jsdoc/api/plugins/doc.js]: there is the stable/unstable distinction though. Is that not what this is handling?

**Comment by ahocevar** [comment]
I feel a bit uncomfortable with this. The `doc` plugin repeats a lot of work that is done by the `api` plugin, and not related to the new `doc` annotation. I'll try to wrap my head around this and see if we can change things so referenced enums, object types and typedefs are always documented.

File: config/jsdoc/api/plugins/doc.js (hunk 1 above)
Comment (ahocevar) [inline config/jsdoc/api/plugins/doc.js]: It is, but I don't think we need to handle that for things that are not part of the API.

Comment (probins) [inline config/jsdoc/api/plugins/doc.js]: I suppose there is the question of what stable/unstable means in the case of typedefs

Comment (probins) [inline config/jsdoc/api/plugins/doc.js]: in which case, maybe we should just remove `stable` from typedefs (and other non-api types)

Comment (ahocevar) [inline config/jsdoc/api/plugins/doc.js]: yes, and they should not show up as "experimental" in the docs.

**Comment by probins** [comment]
ok, so I'll leave that up to you for the moment. If you like, I can remove the `stable` and remove the level logic from the `@doc` plugin

**Comment by ahocevar** [comment]
> ok, so I'll leave that up to you for the moment.

Thanks for your work so far! I'll see if I can find a quick solution. If not, I'd appreciate if you could finish this. But I'll let you know, no later than tomorrow.

**Comment by ahocevar** [comment]
@probins, I was able to implement what I had in mind. See #5427.

**Comment by probins** [comment]
closed in favour of #5427.

**Pull request closed without merging**